### ISSUE TITLE
Encourage removing `{ .. }` from a pattern if enum variant is field-less

### ISSUE THREAD

**Issue by urben1680** [rust-lang/rust-clippy#13400]
### What it does

Match patterns using `Enum::Variant { .. }` should be linted on if the enum variant is field-less.

Example:
```rust
match Some(42) {
    Some(v) => v,
    None { .. } /* <- lint this */ => 1
}
```

### Advantage

- Removes noise from when the vairant used to have fields (before refactoring for example)

### Drawbacks

None as far I can tell.

### Example

```rust
None { .. } => 1
     ^^^^^^
```

Could be written as:

```rust
None => 1
```

**Comment by lolbinarycat** [comment]
> This pattern may still useful in macros so this should still be possible to be ignored with an `[allow(...)]`

1. all lints can be ignored in that way, if they couldn't be, they wouldn't be lints, they would be errors
2. macro generated code is usually not linted
